src/m1_facilitator/models/Message.ts

```diff
@@ -0,0 +1,123 @@
+// Copyright 2020 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+import BigNumber from 'bignumber.js';
+
+import Comparable from '../../m0_facilitator/observer/Comparable';
+
+/** Message types for deposit and withdraw */
+export enum MessageType {
+  Deposit = 'deposit',
+  Withdraw = 'withdraw',
+}
+
+/** Status of messages */
+export enum MessageStatus {
+  Undeclared = 'undeclared',
+  Declared = 'declared',
+}
+
+/**
+ * Represents Message model object.
+ */
+export default class Message extends Comparable<Message> {
+  public messageHash: string;
+
+  public type: MessageType;
+
+  public sourceStatus: MessageStatus;
+
+  public targetStatus: MessageStatus;
+
+  public gatewayAddress: string;
+
+  public intentHash?: string;
+
+  public gasPrice?: BigNumber;
+
+  public gasLimit?: BigNumber;
+
+  public sourceDeclarationBlockNumber?: BigNumber;
+
+  public createdAt?: Date;
+
+  public updatedAt?: Date;
+
+  /**
+   * Constructor to set fields of Message model.
+   *
+   * @param messageHash Message hash is unique for each request.
+   * @param type Type of the message deposit/withdraw.
+   * @param intentHash Intent hash.
+   * @param sourceStatus Status of source.
+   * @param targetStatus Status of target.
+   * @param gasPrice Gas price that depositor/withdrawer is ready to pay to get
+   *                 deposit/withdraw process done.
+   * @param gasLimit Gas limit that depositor/withdrawer is ready to pay.
```

Comment:
Please update the documentation:
`Fee gas price for reward calculation.` &
`Fee gas limit for reward calculation.`

Reply:
Done.